os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -677,18 +681,20 @@ static int i2c_esp32_read_msg(struct i2c_dev_s *dev, uint16_t addr, struct i2c_m
 			*msg->buffer++ = v;
 		}
 		msg->length -= to_read;
+		read_count += to_read;
 
 		i2c_esp32_reset_txfifo(priv->i2c_num);
 	}
 
-	return 0;
+	return read_count;
```

Comment:
please check if we need to return error status for those APIs, e.g.  EIO, or ETIMEOUT

Reply:
Fixed